src/scripting.cpp

```diff
@@ -159,6 +229,46 @@ QJSValue Scripting::dimensions(int width, int height) {
     return obj;
 }
 
+QJSValue Scripting::position(int x, int y) {
+    QJSValue obj = instance->engine->newObject();
+    obj.setProperty("x", x);
+    obj.setProperty("y", y);
+    return obj;
+}
+
+Tile Scripting::toTile(QJSValue obj) {
+    if (!obj.hasProperty("tileId")
+     || !obj.hasProperty("xflip")
+     || !obj.hasProperty("yflip")
+     || !obj.hasProperty("palette")) {
+        return Tile();
+    }
+    Tile tile = Tile();
+    tile.tileId = obj.property("tileId").toInt();
+    tile.xflip = obj.property("xflip").toBool();
+    tile.yflip = obj.property("yflip").toBool();
+    tile.palette = obj.property("palette").toInt();
+    return tile;
+}
+
+QJSValue Scripting::fromTile(Tile tile) {
+    QJSValue obj = instance->engine->newObject();
+    obj.setProperty("tileId", tile.tileId);
+    obj.setProperty("xflip", tile.xflip);
+    obj.setProperty("yflip", tile.yflip);
+    obj.setProperty("palette", tile.palette);
+    return obj;
+}
+
 QJSEngine *Scripting::getEngine() {
     return instance->engine;
 }
+
+QImage Scripting::getImage(QString filepath) {
+    const QImage * image = instance->imageCache.value(filepath, nullptr);
+    if (!image) {
+        image = new QImage(filepath);
```

Comment:
What happens if the image filepath doesn't exist?  (Does it crash porymap?)

Reply:
That's handled by `Overlay::addImage`:
```
if (image.isNull()) {
    logError(QString("Failed to load image '%1'").arg(filepath));
    return false;
}
```

Reply:
(If the filepath doesn't exist it returns a null image, not a null pointer)